helm/resource_helm_release.go

```diff
@@ -736,8 +736,8 @@ func getInstalledReleaseVersion(ctx context.Context, m *Meta, cfg *action.Config
 }
 
 func (r *HelmRelease) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
-	var state HelmReleaseModel
-	diags := req.Plan.Get(ctx, &state)
+	var plan HelmReleaseModel
+	diags := req.Plan.Get(ctx, &plan)
```

Comment:
I renamed this variable to `plan` for clarity, since it is using the plan model.

Reply:
The issue is actually at the bottom of the diff in the `Update()` function. It getting the login credentials from the previously applied state, rather than using the currently fetched credentials.

https://github.com/hashicorp/terraform-provider-helm/pull/1687/files#diff-c548ca21c915d6a4918945eb9741a292966d795ca0a4e9e6dfa4f8c8106f5691L1069-R1069